cloudify_marketplace/resources/cloudify_security_reconfiguration

```diff
@@ -0,0 +1,267 @@
+    logger.info('{action}ing {service}'.format(action=action, service=name))
+    # If it fails to stop that's not a critical problem
+    subprocess.call(['/sbin/service', name, action])
+
+
+class ConfigFile(object):
+    """
+    A context manager for opening and editing config files.
+    Will return a ConfigFile object with attributes:
+        `old`: current file contents
+        `new': will be used to overwrite the config file on exiting the context
+    """
+    def __init__(self, file):
+        self.file = file
+
+    def __enter__(self):
+        with open(self.file) as fh:
+            self.new = self.old = fh.read()
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        with open(self.file, 'w') as fh:
+            fh.write(self.new)
+
+
+def update_config(file, rules):
+    with ConfigFile(file) as config:
+        for pattern, repl in rules:
+            config.new = re.sub(pattern, repl, config.new)
+
+
+def update_riemann_config(user, password):
+    update_config(RIEMANN_CONFIG_PATH,
+                  [('(?<=RABBITMQ_USER=").*(?=")', user),
+                   ('(?<=RABBITMQ_PASS=").*(?=")', password),
+                   ])
+
+
+def update_logstash_config(user, password):
+    update_config(LOGSTASH_CONFIG_PATH,
+                  [('(?<=user => ").*(?=")', user),
+                   ('(?<=password => ").*(?=")', password),
+                   ])
+
+
+def update_amqpinflux_config(user, password):
+    update_config(AMQPINFLUX_CONFIG_PATH,
+                  [('(?<=AMQP_USERNAME=).*', user),
+                   ('(?<=AMQP_PASSWORD=).*', password),
+                   ])
+
+
+def update_mgmtworker_config(user, password):
+    with ConfigFile(MGMTWORKER_CONFIG_PATH) as config:
+        mgmtworker_config = json.loads(config.old)
+        mgmtworker_config['broker_username'] = user
+        mgmtworker_config['broker_password'] = password
+        config.new = json.dumps(mgmtworker_config)
+
+
+def update_restservice_config(user, password):
+    with ConfigFile(RESTSERVICE_CONFIG_PATH) as config:
+        restservice_config = yaml.load(config.old)
+        restservice_config['amqp_username'] = user
+        restservice_config['amqp_password'] = password
+        # Disable default flow style to maintain configish layout
+        config.new = yaml.safe_dump(
+            restservice_config,
+            default_flow_style=False)
+
+
+def update_rabbitmq_credentials(username, password):
+    # delete all rabbit users
+    output = subprocess.check_output([RABBITMQCTL, 'list_users'])
+    # First line says "Listing user ..."
+    for line in output.splitlines()[1:]:
+        user = line.split()[0]
+        subprocess.call([RABBITMQCTL, 'delete_user', user])
+
+    # Add the new rabbit user
+    subprocess.call([RABBITMQCTL, 'add_user', username, password])
+    subprocess.call([RABBITMQCTL, 'set_permissions', username,
+                     '.*', '.*', '.*'])
+
+    service('restart', BROKER_SERVICE_NAME)
```

Comment:
Cool, if it doesn't need a restart for the certs then I agree, let's not restart it. I've put in a reload as you suggest just to be sure, should prevent hiccups from other services :)